modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -372,14 +372,12 @@ public Object visit(PropertyIsLike filter, Object extraData) {
         return output;
     }
 
-    // There is no "NULL" in MongoDB, but I assume that TODO add null support
-    // the non-existence of a column is the same...
     @Override
     public Object visit(PropertyIsNull filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
-
-        String prop = convert(filter.accept(this, null), String.class);
-        output.put(prop, new BasicDBObject("$exists", false));
+        String prop = convert(filter.getExpression().evaluate(null), String.class);
+        // mongodb filter { item: null } supports both: null value and nonexistent attribute
+        output.put(prop, null);
         return output;
```

Comment:
So ``PropertyIsNull`` shoudl yield ``TRUE`` if the value of the property is ``NULL``. Although, in the case of MongoDB, which is a schema less store, since the property may not exists at all, this is interpreted as the property doesn't exists or is ``ǸULL``, looks correct to me :smiley:.

MongoDB documentation link:
https://docs.mongodb.com/manual/tutorial/query-for-null-fields/#equality-filter